### ISSUE TITLE
Set an environment variable in Ruby when setting it in the registry

### ISSUE THREAD

**Issue by ferventcoder** [badgerious/puppet-windows-env#12]
The Puppet process may not be notified of a change in environment variables.

The broadcast may not make it out for every process. https://github.com/chocolatey/puppet-chocolatey/issues/86

**Comment by badgerious** [comment]
Makes sense. I think it might be desirable to *always* set the variable in ruby, even if not changed. You might be running ruby by hand from a prompt (with a stale environment), and would only see the new value when it changed, but subsequent runs would see the old value. That could be confusing.

**Comment by ferventcoder** [comment]
I'd actually prefer an explicit switch. Thoughts?

**Comment by badgerious** [comment]
A switch where 

(option 1)
off = "never touch ruby's ENV"
on =  "always update ENV, even if the registry didn't need updating"

or 

(option 2)
off = "never touch ruby's ENV"
on = "Update ruby's ENV when the registry changes only"

?

I'm not sure I understand the motivation for either. When would you want to keep the old environment?

**Comment by ferventcoder** [comment]
Option 1.

> I'm not sure I understand the motivation for either. When would you want to keep the old environment?

Sometimes you may have shell specific values and you don't want them to be messed with. We've seen this in choco a bit. Say I have an environment variable for PUPPET_VERSION set to 3.8.3 (as a default). Then I override it in the shell for testing. Now if I don't have an explicit option to turn off the behavior of always setting the environment vars based on the registry, I'm in trouble.

This isn't exactly a 1:1 use case here, but there are likely times when something similar is true.

**Comment by ferventcoder** [comment]
I'd mostly assume that the environment variables may not be up to date, and thus seek option 1 for always setting them.

**Comment by badgerious** [comment]
OK, I buy that.

**Comment by rismoney** [comment]
Is it possible to get traction on this one?  I believe most of the changes were in the develop branch...
